odoo/service/db.py

```diff
@@ -361,14 +361,7 @@ def list_dbs(force=False):
     db = odoo.sql_db.db_connect('postgres')
     with closing(db.cursor()) as cr:
         try:
-            db_user = odoo.tools.config["db_user"]
-            if not db_user and os.name == 'posix':
-                import pwd
-                db_user = pwd.getpwuid(os.getuid())[0]
-            if not db_user:
-                cr.execute("select usename from pg_user where usesysid=(select datdba from pg_database where datname=%s)", (odoo.tools.config["db_name"],))
-                res = cr.fetchone()
-                db_user = res and str(res[0])
+            db_user = odoo.tools.config["db_user"] or cr._cnx.get_dsn_parameters().get('user')
```

Comment:
Why the first part of the `or`? Doesn't the second part also return `config['db_user']` if there is one?

Reply:
You're right (as most of the time :+1: ) I cannot see another case where the cursor user could be different than the config['db_user']